CIP-0042/README.md

```diff
@@ -0,0 +1,115 @@
+`secp256k1-haskell`; we need to do this because `secp256k1-haskell` does not
+currently have support for Schnorr signatures at all.
+
+The binds would be implemented in `cardano-base`, with new builtins in Plutus
+Core on the basis of those binds.
+
+The builtins would be costed as follows: ECDSA signature verification has
+constant cost, as the message, verification key and signature are all
+fixed-width; Schnorr signature verification is instead linear in the message
+length, as this can be arbitrary, but as the length of the verification key and
+signature are constant, the costing will be constant in both.
+
+## Rationale
+
+Implementing these functions as direct builtins in Plutus Core permits a
+workflow involving interoperability services like Wanchain and Renbridge. An
+example of such a workflow would be: given some Cardano assets, their Bitcoin
+equivalent could be minted, but only if the correct Schnorr signature can be
+provided. 
+
+We consider the implementation trustworthy: `libsecp256k1` is the reference
+implementation for both signature schemes, and is already being used in
+production by Bitcoin. `secp256k1-haskell` is a library of fairly low-level and
+thin wrappers around the functionality provided by `libsecp256k1`: its only
+purpose is convenience, and it provides no new functionality.
+
+An alternative approach could be to provide low-level primitives, which would
+allow any signature scheme (not just the ones under consideration here) to be
+implemented by whoever needs them. While this approach is certainly more
+flexible, it has two significant drawbacks:
+
+* It requires 'rolling your own crypto', rather than re-using existing
+  implementations. This has been shown historically to be a bad idea;
+  furthermore, if existing implementations have undergone review and audit, and
+  such re-implementations would not benefit.
+* It would be significantly costlier, as the computation would happen in Plutus
+  Core. Given the significant on-chain size restrictions, this would likely be
+  too costly for general use: many such schemes rely on large precomputed
+  tables, for example, which are totally unviable on-chain.
+
+It may be possible that some set of primitive can avoid both of these issues
+(for example, the suggestions in [this
+CIP](https://github.com/cardano-foundation/CIPs/pull/220)); in the meantime,
+providing direct support for commonly-used schemes such as these is worthwhile.
+
+## Backward Compatibility
+
+At the Plutus Core level, implementing this proposal induces no
+backwards-incompatibility: the proposed new primitives do not break any existing
+functionality or affect any other builtins. Likewise, at levels above Plutus
+Core (such as `PlutusTx`), no existing functionality should be affected.
+
+On-chain, this requires a hard fork: the upcoming July hardfork [already takes
+this feature into account](https://www.youtube.com/watch?v=B0tojqvMgB0&t=1148s).
+
+## Implementations
```

Comment:
Let's make this "Path to active", and point out that the costing work will be done by the Plutus Core team.

Reply:
Done.